api/src/main/java/io/opentelemetry/trace/attributes/IntOrStringAttributeSetter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace.attributes;
+
+import io.opentelemetry.trace.Span;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Defines the behavior for a span attribute with long values which are normally handled as integers
+ * or strings in most applications.
+ */
+@Immutable
+public class IntOrStringAttributeSetter extends AbstractAttributeSetter<Integer> {
```

Comment:
What about this: We remove this class but instead give LongAttributeSetter two methods:

* trySetParsed(String) which tries to parse the argument and does nothing on failure.
* setParsedOrRaw(String) which tries to parse the argument and sets the string as-is on failure
* remove set(String) from to avoid *accidentally* passing a string

I would think that this is much cleaner than having this IntOrStringAttributeSetter  where it is not clear what set does on failure.